[새리]미션 2: 체스판 생성

sallyjellyy opened the pull request:
안녕하세요, 2021 마스터즈 백엔드반 새리입니다.
제 실력에는 요구사항을 최소한으로 구현하는 편이 나중에 기능을 추가로 구현할 때, 또는 코드를 수정할 때 편리할 것 같아 그렇게 하였습니다.
보드 테스트 클래스에 중복되는 부분을 줄이는 과정에서도 약간 어려움이 있었지만 제 나름대로 구현해보았습니다.
시간 내주어 리뷰해주셔서 감사합니다!

sallyjellyy committed:
feat: Add test case that checks default constructor on Pawn class

sallyjellyy committed:
feat: Create the default constructor for Pawn class

- The default constuctor sets pawn's color to "white"

sallyjellyy committed:
refactor: Use constants on color

- Default constructor takes the constant value WHITE

sallyjellyy committed:
feat: Create Board class and add findPawn, getPawnSize, addPawn methods

sallyjellyy committed:
feat: Create test to see if Board class works properly

- Create andn add pawn to the board and check if the methods in Board class works properly

sallyjellyy committed:
refactor: Change package to pieces

- Pawn's constructor that takes String type of color as parameter caused compile error since the constructor was not public.

sallyjellyy committed:
refactor: Add setUp method and  verfyBoard method to cut duplicates of codes

sallyjellyy committed:
refactor: Change color varible non-final to final

sallyjellyy committed:
Merge branch 'step2' into step1

SANGCO commented on src/test/java/chess/BoardTest.java:
```diff
@@ -0,0 +1,30 @@
+        assertThat(board.getPawnSize()).isEqualTo(index + 1);
+        assertThat(board.findPawn(index)).isEqualTo(pawn);
+    }
+}
```
![image](https://user-images.githubusercontent.com/26179444/107593824-b59d4380-6c53-11eb-96a2-8ce5bd465118.png)
IntelliJ 설정을 찾아서 적용해 보아요.
[How to make IntelliJ IDEA insert a new line at every end of file?](https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file)
[파일 끝에 개행을 추가해야 하는 이유](https://blog.coderifleman.com/2015/04/04/text-files-end-with-a-newline/)

sallyjellyy replied:
파일간의 차이를 알기 위해 개행을 해주어야 하는군요! 감사합니다. 설정 고쳤습니다.

SANGCO commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.List;
+
+public class Board {
+    List<Pawn> pawns = new ArrayList<>();
+
+    void addPawn(Pawn pawn) {
+        pawns.add(pawn);
+    }
+
+    int getPawnSize() {
+        return pawns.size();
+    }
+
+    Pawn findPawn(int i) {
+        return pawns.get(i);
+    }
```
접근 제어자에 대해서 다시 학습을 해보시고 
Board 클래스 필드, 메서드 접근 제어자를 수정해 봅시다!
질문 있으시면 댓글 남겨주세요~

SANGCO commented on src/main/java/pieces/Pawn.java:
```diff
@@ -0,0 +1,20 @@
+
+    private final String color;
+
+    Pawn() {
```
여기도 접근 제어자에 대해서 동일한 피드백 입니다~

SANGCO commented on src/test/java/pieces/PawnTest.java:
```diff
@@ -1,12 +1,21 @@
 
+    @Test
+    @DisplayName("폰 컬러의 디폴트값은 white이다")
+    public void create_기본생성자() throws Exception {
```
접근 제어자 동일한 피드백 입니다.

sallyjellyy replied:
접근제어자 관련 피드백 받은 부분 모두 수정하였습니다!

SANGCO commented on src/test/java/chess/BoardTest.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Test
+    @DisplayName("폰이 보드에 정상적으로 등록")
+    public void create() throws Exception {
+        verifyBoard(board, whitePawn, 0);
+        verifyBoard(board, blackPawn, 1);
+    }
+
+    void verifyBoard(Board board, Pawn pawn, int index) {
```
public 접근 제한자를 제거해도 좋을거 같아요.
JUnit5에서는 테스트 클래스나 @test가 붙은 메서드에 public을 붙이지 않아도 된답니다.
클래스 내에서만 사용하는 메서드의 접근제어자는 private이 좋겠죠?

sallyjellyy replied:
테스트메서드에는 퍼블릭을 붙이지 않아도 되는군요! 감사합니다! 수정했습니다.

SANGCO commented on src/test/java/chess/BoardTest.java:
```diff
@@ -0,0 +1,30 @@
+
+public class BoardTest {
+
+    Board board;
+    Pawn whitePawn, blackPawn;
```
이 부분도 접근 제어자에 대해서 고민해 봅시다!

SANGCO commented on src/main/java/chess/Board.java:
```diff
@@ -0,0 +1,22 @@
+package chess;
```
패키지를 전체적으로 한번 정리를 해야할거 같습니다.
net.min27604.chess 이런식으로 만들어서 그 하위에 폴더와 파일들을 만들어 봅시다.
메인 클래스도 본인이 만든 패키지로 옮기면 net.honux.chess는 이제 필요 없으니 지우는게 좋겠죠?

sallyjellyy replied:
넵 안 그래도 net.honux.chess부분을 남겨둬야하나 고민했었는데 피드백 감사합니다. net.sally 패키지를 따로 만들어 기존에 있던 패키지들을 다 정리했습니다.

SANGCO reviewed (changes requested):
새리님 안녕하세요, 리뷰어 상코입니다.
작성하신 코드에 대해서 코멘트를 남겼습니다. 
확인해 보시고 질문 있으시면 댓글 부탁드립니다~

sallyjellyy committed:
refactor: Change directory, fix access modifier

- Deleted net.honux.chess package since it's unnecessary in this project. Redirected all the packages into net.sally.
- Changed all the test method's access modifier to package private and all other methods in test classes to private because those won't be used in any other classes.
- Changed all the methods in Board class and Pawn class to public to use them in other classes but all the member variables to private.

sallyjellyy committed:
refactor: Changed verifyPawn() method to private

- veryfyPawn() method is only used within the PawnTest class

SANGCO reviewed (approved):
수정 잘하셨네요~
다음 미션도 화이팅 화이팅
